### PR TITLE
copr: Add logic to detect correct reposdir location and override detected copr chroot support

### PR DESCRIPTION
**Pull request by Conan-Kudo**
The `copr` plugin currently assumes that repos are installed to `/etc/yum.repos.d`. This is a problem as DNF itself can have repositories in `/etc/yum.repos.d`, `/etc/yum/repos.d`, or `/etc/distro.repos.d` by default.

We also expect that Fedora 25 will switch to using `/etc/distro.repos.d`, and Mageia will be using this path in Mageia 6. Even if this wasn't the case, it's a good idea to detect where the reposdir actually is and use that location.

Additionally, derivative distributions (such as Korora) need a way to override the detected chroots, so that they can use the parent distribution's chroots properly.

### CODE REVIEW

**Comment by jsilhan** [comment]
LGTM. We probably need to do something similar in `config-manager` plugin.

**Comment by Conan-Kudo** [comment]
@jsilhan I actually stole the function for detecting reposdir from config-manager. ;)

File: plugins/copr.py [hunk 1]
```diff
@@ -45,6 +47,29 @@
 else:
     from ConfigParser import ConfigParser
 
+def get_reposdir():
+    myrepodir = None
+    # put repo file into first reposdir which exists or create it
+    for rdir in self.base.conf.reposdir:
```
Comment (xsuchy) [inline plugins/copr.py]: self is not defined here.

File: plugins/copr.py [hunk 2]
```diff
@@ -45,6 +47,29 @@
...
+        if os.path.exists(rdir):
+            myrepodir = rdir
+
+        if not myrepodir:
+            myrepodir = self.base.conf.reposdir[0]
```
Comment (xsuchy) [inline plugins/copr.py]: again. self is not defined

**Comment by Conan-Kudo** [comment]
@xsuchy I think I fixed that now.

**Comment by xsuchy** [comment]
And there is some PEP8 formating warnings (try to run python3-pylint on this module). But this is is minor and I can actually fix it myself later. But please fix that issue with self variable.

**Comment by Conan-Kudo** [comment]
@xsuchy Fixed in latest amended commit.

File: plugins/copr.py [hunk 3]
```diff
@@ -79,6 +81,29 @@ class CoprCommand(dnf.cli.Command):
   copr search tests
     """)
 
+# Useful for forcing a distribution
+    def _chroot_config(self):
+        cp = self.read_config(self.base.conf, PLUGIN_CONF)
```
Comment (xsuchy) [inline plugins/copr.py]: File "/usr/lib/python3.4/site-packages/dnf-plugins/copr.py", line 86, in _chroot_config
    cp = self.read_config(self.base.conf, PLUGIN_CONF)
AttributeError: 'CoprCommand' object has no attribute 'read_config'

File: plugins/copr.py [hunk 4]
```diff
@@ -240,9 +265,11 @@ def _need_root(cls):
 
     @classmethod
     def _guess_chroot(cls):
-        """ Guess which choot is equivalent to this machine """
+        """ Guess which chroot is equivalent to this machine """
         # FIXME Copr should generate non-specific arch repo
-        dist = platform.linux_distribution()
+        dist = self._chroot_config()
```
Comment (xsuchy) [inline plugins/copr.py]: here is reference to self, while this method has cls.

**Comment by Conan-Kudo** [comment]
@xsuchy I moved the method into the config() hook and adjusted accordingly. hopefully it should work...

**Comment by jsilhan** [comment]
> @jsilhan I actually stole the function for detecting reposdir from config-manager. ;)

Then it would be a good idea to move it into `dnfcorelib` to not duplicate the code.

File: plugins/copr.py [hunk 5]
```diff
@@ -240,9 +267,11 @@ def _need_root(cls):
 
     @classmethod
     def _guess_chroot(cls):
-        """ Guess which choot is equivalent to this machine """
+        """ Guess which chroot is equivalent to this machine """
         # FIXME Copr should generate non-specific arch repo
-        dist = platform.linux_distribution()
+        dist = CHROOT_CONFIG
```
Comment (xsuchy) [inline plugins/copr.py]: I do not like the usage of global variable. Can you make CHROOT_CONFIG class attribute and refer to it as self.cli.CHROOT_CONFIG (and rename cls to self)?

File: plugins/copr.py [hunk 6]
```diff
@@ -56,6 +60,17 @@ def __init__(self, base, cli):
         if cli is not None:
             cli.register_command(CoprCommand)
 
+    # Useful for forcing a distribution
+    def config(self):
+        cp = self.read_config(self.base.conf, PLUGIN_CONF)
```
Comment (xsuchy) [inline plugins/copr.py]: File "/usr/lib/python3.4/site-packages/dnf-plugins/copr.py", line 65, in config
    cp = self.read_config(self.base.conf, PLUGIN_CONF)
AttributeError: 'Copr' object has no attribute 'base'

Please before you push another commit, do "tito --build --rpm -i; dnf copr enable fooo/bar" and test if it really works.

**Pull request closed without merging**